components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/definitions/APIDefinitionFromSwagger12.java

```diff
@@ -0,0 +1,418 @@
+                                    Scope scope = new Scope();
+                                    scope.setKey((String) scopeMap.get("key"));
+                                    scope.setName((String) scopeMap.get("name"));
+                                    scope.setRoles((String) scopeMap.get("roles"));
+                                    scope.setDescription((String) scopeMap.get("description"));
+                                    scopeList.add(scope);
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        } catch (ParseException e) {
+            handleException("Invalid resource configuration ", e);
+        }
+
+        return scopeList;
+    }
+
+    @Override
+    @Deprecated
+    public void saveAPIDefinition(API api, String apiDefinitionJSON, Registry registry) throws APIManagementException {
+        String apiName = api.getId().getApiName();
+        String apiVersion = api.getId().getVersion();
+        String apiProviderName = api.getId().getProviderName();
+
+        try {
+            String resourcePath = APIUtil.getSwagger12DefinitionFilePath(apiName, apiVersion, apiProviderName);
+            resourcePath = resourcePath + APIConstants.API_DOC_1_2_RESOURCE_NAME;
+            Resource resource = registry.newResource();
+
+            resource.setContent(apiDefinitionJSON);
+            resource.setMediaType("application/json");
+            registry.put(resourcePath, resource);
+
+            //Set visibility as same as the API visibility
+            APIUtil.setResourcePermissions(apiProviderName, api.getVisibility(), null, resourcePath);
+
+        } catch (RegistryException e) {
+            handleException("Error while adding Swagger Definition for " + apiName + "-" + apiVersion, e);
+        } catch (APIManagementException e) {
+            handleException("Error while adding Swagger Definition for " + apiName + "-" + apiVersion, e);
+        }
+    }
+
+    @Override
+    @Deprecated
+    public String getAPIDefinition(APIIdentifier apiIdentifier, Registry registry) throws APIManagementException {
+        String resourcePath = APIUtil.getSwagger12DefinitionFilePath(apiIdentifier.getApiName(),
+                apiIdentifier.getVersion(), apiIdentifier.getProviderName());
+
+        JSONParser parser = new JSONParser();
+        JSONObject apiJSON = null;
+        try {
+            if (registry.resourceExists(resourcePath + APIConstants.API_DOC_1_2_RESOURCE_NAME)) {
+                Resource apiDocResource = registry.get(resourcePath + APIConstants.API_DOC_1_2_RESOURCE_NAME);
+                String apiDocContent = new String((byte[]) apiDocResource.getContent());
+                apiJSON = (JSONObject) parser.parse(apiDocContent);
+                JSONArray pathConfigs = (JSONArray) apiJSON.get("apis");
+
+                for (int k = 0; k < pathConfigs.size(); k++) {
```

Comment:
APIDefinitionFromSwagger12 is used with swagger v1.2(In APIM 1.8.0) and removed since it is no longer in use(In APIM 1.9.0)
Instead we have APIDefinitionFromSwagger20 Implemented from same abstract class which implements APIDefinitionFromSwagger12